virt-v2v/cold/entrypoint.go

```diff
@@ -196,47 +212,72 @@ func executeVirtV2v(args []string) (err error) {
 		return
 	}
 
+	if source == OVA {
+		scanner := bufio.NewScanner(teeErr)
+		const maxCapacity = 1024 * 1024
+		buf := make([]byte, 0, 64*1024)
+		scanner.Buffer(buf, maxCapacity)
+
+		for scanner.Scan() {
+			line := scanner.Bytes()
+			if match := UEFI_RE.FindSubmatch(line); match != nil {
+				fmt.Println("UEFI firmware detected")
+				firmware = "efi"
+			}
+		}
+
+		if err = scanner.Err(); err != nil {
+			fmt.Println("Output query failed:", err)
+			return err
+		}
+	}
+
 	if err = virtV2vCmd.Wait(); err != nil {
 		fmt.Printf("Error waiting for virt-v2v to finish: %v\n", err)
 		return
 	}
 	return
 }
 
-func getXMLFile(dir, fileExtension string) (string, error) {
+func getYamlFile(dir, fileExtension string) (string, error) {
 	files, err := filepath.Glob(filepath.Join(dir, "*."+fileExtension))
 	if err != nil {
 		return "", err
 	}
 	if len(files) > 0 {
 		return files[0], nil
 	}
-	return "", fmt.Errorf("XML file was not found.")
+	return "", fmt.Errorf("yaml file was not found")
 }
 
-func ovfHandler(w http.ResponseWriter, r *http.Request) {
-	if xmlFilePath == "" {
-		fmt.Println("Error: XML file path is empty.")
-		http.Error(w, "XML file path is empty", http.StatusInternalServerError)
+func vmHandler(w http.ResponseWriter, r *http.Request) {
+	if yamlFilePath == "" {
+		fmt.Println("Error: YAML file path is empty.")
+		http.Error(w, "YAML file path is empty", http.StatusInternalServerError)
 		return
 	}
 
-	xmlData, err := os.ReadFile(xmlFilePath)
+	err := addFirmwareToYaml(yamlFilePath)
+	if err != nil {
+		fmt.Printf("Error setting yaml file: %v\n", err)
```

Comment:
setting yaml file? you mean modify/change the yaml file (or vm configuration), no?